### PR TITLE
Include minor fixes identified from demo

### PR DESCRIPTION
**Pull request by lewijacn**
### Description
This change includes minor fixes that were identified when putting pieces together for a demo. Additional pieces that will be added elsewhere to get complete setup shown in demo include:
* Jupyter container CDK setup
* MSK/Kafka cluster deployed to AWS
* Zero Latency fix (to be added with/after PR for E2E from Greg)

### Issues Resolved
None

### Testing
Manual testing

### Check List
- [ ] New functionality includes testing
  - [ ] All tests pass, including unit test, integration test and doctest
- [X] New functionality has been documented
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: TrafficCapture/KafkaPrinter/src/main/java/org/opensearch/KafkaPrinter.java [hunk 1]
```diff
@@ -58,6 +58,11 @@ public static void main(String[] args) {
         properties.setProperty("value.deserializer", "org.apache.kafka.common.serialization.ByteArrayDeserializer");
         properties.setProperty(ConsumerConfig.GROUP_ID_CONFIG, groupId);
         properties.setProperty(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+        // Required for using SASL auth with MSK public endpoint
+        //properties.setProperty("security.protocol", "SASL_SSL");
+        //properties.setProperty("sasl.mechanism", "AWS_MSK_IAM");
+        //properties.setProperty("sasl.jaas.config", "software.amazon.msk.auth.iam.IAMLoginModule required;");
+        //properties.setProperty("sasl.client.callback.handler.class", "software.amazon.msk.auth.iam.IAMClientCallbackHandler");
```
Comment (okhasawn) [inline TrafficCapture/KafkaPrinter/src/main/java/org/opensearch/KafkaPrinter.java:65]: I guess this is the zero-latency fix that you're going to uncomment after Greg's E2E PR is merged?

**Review by okhasawn — COMMENTED** [review]
LGTM. Left a quick question on the commented out code though.

File: deployment/docker/kafka-puller/run_app.sh [hunk 2]
```diff
@@ -1,14 +1,23 @@
...
 do
-  java -cp build/libs/KafkaPrinter.jar:"$jars" org.opensearch.KafkaPrinter -b "$bootstrap_server" -t "$topic_name" | nc -v localhost 9210
+
+  # Verify netcat can connect to Replayer before process begins
+  while ! nc -v -w 1 localhost 9210
+  do
+    >&2 echo "Attempting to connect to Traffic Replayer..."
+    sleep 2
+  done
+  sleep 30
```
Comment (gregschohn) [inline deployment/docker/kafka-puller/run_app.sh:18]: I don't understand this code.  Is there a way that you can do dependencies in Fargate?  You can (& should) in compose definitions.

Comment (gregschohn) [reply deployment/docker/kafka-puller/run_app.sh:18]: Why is this sleep 30 seconds and the one below 7?  Why do you need any sleeps at all?

**Review by okhasawn — APPROVED** [review]

**Comment by okhasawn** [comment]
Which changes address this point "Jupyter container CDK setup"?

File: deployment/docker/traffic-replayer/run_app.sh [hunk 3]
```diff
@@ -10,22 +10,22 @@ do
...
   # Allow time for TC to begin listening after terminated connection
-  sleep 10
+  sleep 7
 
   touch triples.log
 
   # These commands look a bit backwards but ultimately will receive input from CW-Puller via netcat and then
   # process with the Replayer jar before ultimately printing the produced triples to stdout via tee and sending to TC via netcat
   # Note: Unbuffer has not been verified to actually be effective here. stdbuf was ruled out for its caveat with tee (https://linux.die.net/man/1/stdbuf)
   unbuffer tail -f triples.log | tee /dev/stderr | nc -v localhost 9220 &
```
Comment (gregschohn) [inline deployment/docker/traffic-replayer/run_app.sh:25]: Why are you using a file at all here?  Why not pipe directly?  Does unbuffer matter for the case where lots of messages are going through?

File: deployment/docker/traffic-replayer/run_app.sh [hunk 4]
```diff
@@ -10,22 +10,22 @@ do
...
-  nc -v -l -p 9210 | java -jar trafficReplayer-uber.jar -o triples.log "$target_endpoint"
+  nc -v -l -p 9210 | java -jar trafficReplayer-uber.jar "$target_endpoint" -o triples.log --insecure
```
Comment (gregschohn) [inline deployment/docker/traffic-replayer/run_app.sh:26]: are you unbuffering the output to triples.log from the java process?

File: TrafficCapture/KafkaPrinter/src/main/java/org/opensearch/KafkaPrinter.java (hunk 1 above)
Comment (lewijacn) [reply TrafficCapture/KafkaPrinter/src/main/java/org/opensearch/KafkaPrinter.java:65]: This is required properties to be able to communicate with an MSK public endpoint. By default I want to allow all kafka clusters but want to keep settings to be able to communicate with MSK if needed

**Comment by lewijacn** [comment]
> Which changes address this point "Jupyter container CDK setup"?

We don't have a PR yet but we need to do some reconfiguring of the jupyter notebook container so that it is easier to get setup and run and will be a followup PR sometime

File: deployment/docker/traffic-replayer/run_app.sh (hunk 4 above)
Comment (gregschohn) [reply deployment/docker/traffic-replayer/run_app.sh:26]: What happens if the uberjar has multiple mains() defined within different source jars?

**Review by gregschohn — APPROVED** [review]
I'm approving this, but it does feel like this might be better on a branch or separate patch.  The CDK changes have long term value (though the https->http one is problematic).  The other stuff seems like it might be better on a branch - but I know that the kafka puller and containers are in need of rework anyway.

File: TrafficCapture/KafkaPrinter/src/main/java/org/opensearch/KafkaPrinter.java (hunk 1 above)
Comment (okhasawn) [reply TrafficCapture/KafkaPrinter/src/main/java/org/opensearch/KafkaPrinter.java:65]: Oh I see, I misunderstood the description of the PR which made me get to this assumption. I apologize.

File: deployment/docker/kafka-puller/run_app.sh (hunk 2 above)
Comment (lewijacn) [reply deployment/docker/kafka-puller/run_app.sh:18]: This is ugly code and needs to be reworked in future change. Sleeps here are a "hack" to allow time for the replayer and comparator to start up

File: deployment/cdk/opensearch-service-migration/lib/migration-assistance-stack.ts [hunk 5]
```diff
@@ -111,7 +119,7 @@ export class MigrationAssistanceStack extends Stack {
             image: ContainerImage.fromDockerImageAsset(trafficReplayerImage),
             // Add in region and stage
             containerName: "traffic-replayer",
-            environment: {"TARGET_CLUSTER_ENDPOINT": "https://" + props.targetEndpoint + ":80"},
+            environment: {"TARGET_CLUSTER_ENDPOINT": "http://" + props.targetEndpoint + ":80"},
```
[the remark this answers is not in the document]
Comment (lewijacn) [reply deployment/cdk/opensearch-service-migration/lib/migration-assistance-stack.ts:122]: Yeah currently we are building Domains that allow http requests but this is not the default we will want long term

File: deployment/docker/traffic-replayer/run_app.sh (hunk 3 above)
Comment (lewijacn) [reply deployment/docker/traffic-replayer/run_app.sh:25]: I believe at the time we were running into issue with having netcat receiving and sending in the same line. That may not be present now, but ideally we move away from netcat anyway

File: TrafficCapture/KafkaPrinter/src/main/java/org/opensearch/KafkaPrinter.java (hunk 1 above)
Comment (lewijacn) [reply TrafficCapture/KafkaPrinter/src/main/java/org/opensearch/KafkaPrinter.java:65]: Yes ideally I would like to have a config file here

File: deployment/docker/traffic-replayer/run_app.sh (hunk 4 above)
Comment (lewijacn) [reply deployment/docker/traffic-replayer/run_app.sh:26]: We were using unbuffer as an alternative to stdbuf as it didn't work with tee commands. That said I don't know if we ever confirmed it was useful here and could possibly be removed. 

I've not ran into that issue, but goal is to move away from uberjar here.

**Pull request merged**